sklearn/impute/_iterative.py

```diff
@@ -402,7 +414,10 @@ def _impute_one_feature(
             )
 
         # update the feature
-        X_filled[missing_row_mask, feat_idx] = imputed_values
+        if hasattr(self, "iloc"):
+            X_filled.iloc[missing_row_mask, feat_idx] = imputed_values
+        else:
+            X_filled[missing_row_mask, feat_idx] = imputed_values
```

Comment:
Would it be worth to implement a `_safe_assign` function since this pattern seems to be repeated in many places ?